crates/codegen/parser/runtime/src/support/parser_result.rs

```diff
@@ -93,7 +93,7 @@ impl Match {
     pub fn is_full_recursive(&self) -> bool {
         self.nodes
             .iter()
-            .flat_map(cst::Node::cursor)
+            .flat_map(|node| cst::Node::create_cursor(node, Default::default()))
```

Comment:
I'm not a fan of using the Default as parameter and think it'd be clearer if we have a separate `cursor_without_offset` (without the argument) function `cursor_with_offset` (or something along the lines).

This is fine for now, but I'd rethink how we can design the API to better convey that only the root node cursor has absolute offsets and other created ones are relative.

Reply:
I think `cursor_without_offset()` should create a cursor that doesn't track offsets at all (for perf, and also to not expose an invalid `.offset` property).
Also I thought about introducing a `TextIndex::ZERO` helper.
But I think we should consider this as part of future updates to the API. I want to keep this PR minimal to address existing feedback.